build/builder.sh

```diff
@@ -6,7 +6,7 @@ image="cockroachdb/builder"
 
 function init() {
   docker build --tag="${image}" - <<EOF
-FROM golang:1.4.2
+FROM golang:1.5
```

Comment:
shouldn't that still work correctly? we always `go install [...]/build-cache`, so the version it was compiled with should be incremented before the `build-cache restore` invocation. I'd expect that to correctly invalidate the cache